modules/xmpp/SignalingLayerImpl.js

```diff
@@ -142,12 +142,14 @@ export default class SignalingLayerImpl extends SignalingLayer {
                 = this._remoteSourceState[endpointId] || (this._remoteSourceState[endpointId] = {});
 
             for (const sourceName of Object.keys(sourceInfoJSON)) {
+                let sourceChanged = false;
```

Comment:
General comment on this file. Can you pl make sure Spot screen sharing won't break with these changes?